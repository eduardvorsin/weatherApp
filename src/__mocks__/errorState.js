import fetchingStatuses from '../redux/fetchingStatuses';

const errorData = {
  data: {},
  loadingStatus: fetchingStatuses.error,
  error: 'mock error',
};

const errorState = {
  weather: {
    hourlyWeather: errorData,
    currentWeather: errorData,
    dailyWeather: errorData,
    humidity: errorData,
    wind: errorData,
  },
  geocoding: errorData,
};

export default errorState;
