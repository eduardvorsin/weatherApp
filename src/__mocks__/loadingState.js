import fetchingStatuses from '../redux/fetchingStatuses';

const loadingData = {
  data: {},
  loadingStatus: fetchingStatuses.loading,
  error: null,
};

const loadingState = {
  weather: {
    hourlyWeather: loadingData,
    currentWeather: loadingData,
    dailyWeather: loadingData,
    humidity: loadingData,
    wind: loadingData,
  },
  geocoding: loadingData,
};

export default loadingState;
