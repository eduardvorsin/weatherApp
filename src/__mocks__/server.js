import { rest } from 'msw';
import { setupServer } from 'msw/node';
import mockGeocodingError from './geocodingError';
import mockWeatherError from './weatherError';

const isCurrentWeatherOrWindRequest = (searchParams) => (
  searchParams.get('current_weather') === 'true'
);

const isHourlyWeatherRequest = (searchParams) => {
  const currentParams = searchParams.getAll('hourly');
  return currentParams.includes('temperature_2m') && currentParams.includes('weathercode');
};

const isDailyWeatherRequest = (searchParams) => {
  const currentParams = searchParams.getAll('daily');
  return currentParams.includes('temperature_2m_max') && currentParams.includes('temperature_2m_min') && currentParams.includes('weathercode');
};

const isHumidityRequest = (searchParams) => {
  const currentParams = searchParams.getAll('hourly');
  return currentParams.includes('apparent_temperature') && currentParams.includes('relativehumidity_2m');
};

const currentWeatherData = {
  current_weather: {
    weathercode: 1,
    temperature: -6.7,
    winddirection: 128,
    windspeed: 1.64,
  },
};
const hourlyWeatherData = {
  hourly: {
    time: [
      '2022-12-27T03:00',
      '2022-12-27T04:00',
      '2022-12-27T05:00',
      '2022-12-27T06:00',
      '2022-12-27T07:00',
      '2022-12-27T08:00',
      '2022-12-27T09:00',
      '2022-12-27T10:00',
      '2022-12-27T11:00',
      '2022-12-27T12:00',
      '2022-12-27T13:00',
      '2022-12-27T14:00',
      '2022-12-27T15:00',
      '2022-12-27T16:00',
      '2022-12-27T17:00',
      '2022-12-27T18:00',
      '2022-12-27T19:00',
      '2022-12-27T20:00',
      '2022-12-27T21:00',
      '2022-12-27T22:00',
      '2022-12-27T23:00',
      '2022-12-28T00:00',
      '2022-12-28T01:00',
      '2022-12-28T02:00',
      '2022-12-28T03:00',
      '2022-12-28T04:00',
      '2022-12-28T05:00',
      '2022-12-28T06:00',
      '2022-12-28T07:00',
      '2022-12-28T08:00',
    ],
    temperature_2m: [
      -4.5, -4.4, -4.3, -4.5, -5.1, -5.6, -5.8, -6, -6.5, -6.8, -7, -6.4,
      -5.3, -4.3, -3.6, -3.4, -3.7, -4.4, -4.9, -5.3, -5.6, -5.8, -5.7, -5.5,
      -5.0, -4.5, -3.9, -3.1, -3.4, -4.9,
    ],
    weathercode: [1, 2, 3, 71, 71, 3, 3, 2, 2, 2, 1, 3, 71, 86, 1, 1, 3, 2, 2,
      71, 71, 3, 2, 1, 71, 2, 3, 1, 2, 71],
  },
};
const dailyWeatherData = {
  daily: {
    time: ['2022-12-26', '2022-12-27', '2022-12-29', '2022-12-30', '2022-12-31', '2023-01-01', '2023-01-02'],
    temperature_2m_max: [-9.4, -8, -7.3, -5, -6.7, -7.6, -9.8],
    temperature_2m_min: [-17.1, -15.5, -16.4, -17.5, -15.7, -16.3, -13.6],
    weathercode: [3, 2, 71, 2, 2, 3, 71],
  },
};
const humidityData = {
  hourly: {
    time: [
      '2022-12-27T00:00',
      '2022-12-27T01:00',
      '2022-12-27T02:00',
      '2022-12-27T03:00',
      '2022-12-27T04:00',
      '2022-12-27T05:00',
      '2022-12-27T06:00',
      '2022-12-27T07:00',
      '2022-12-27T08:00',
      '2022-12-27T09:00',
    ],
    apparent_temperature: [-17.4, -18, -18.6, -19.1, -19.5, -19, -17, -16, -15, 14],
    relativehumidity_2m: [81, 82, 83, 84, 85, 86, 71, 72, 73, 74],
  },
};
const geocodingData = {
  address: {
    city: 'Вестминстер',
    state_district: 'Большой Лондон',
    state: 'Англия',
    country: 'Великобритания',
    country_code: 'gb',
    'ISO3166-2-lvl4': 'GB-ENG',
    'ISO3166-2-lvl8': 'GB-WSM',
  },
};

const handlers = [
  rest.get('https://api.open-meteo.com/v1/forecast', (req, res, ctx) => {
    const { searchParams } = req.url;

    if (isCurrentWeatherOrWindRequest(searchParams)) {
      return res(ctx.json(currentWeatherData), ctx.delay(150));
    }
    if (isHourlyWeatherRequest(searchParams)) {
      return res(ctx.json(hourlyWeatherData), ctx.delay(150));
    }
    if (isDailyWeatherRequest(searchParams)) {
      return res(ctx.json(dailyWeatherData), ctx.delay(150));
    }
    if (isHumidityRequest(searchParams)) {
      return res(ctx.json(humidityData), ctx.delay(150));
    }

    return res(ctx.json('unknown request'), ctx.delay(150));
  }),
  rest.get('https://nominatim.openstreetmap.org/reverse', (_, res, ctx) => (
    res(ctx.json(geocodingData), ctx.delay(150))
  )),
];

export const errorHandlers = [
  rest.get('https://nominatim.openstreetmap.org/reverse', (_, res, ctx) => res(ctx.status(404), ctx.json(mockGeocodingError), ctx.delay(150))),
  rest.get('https://api.open-meteo.com/v1/forecast', (_, res, ctx) => res(ctx.status(404), ctx.json(mockWeatherError), ctx.delay(150))),
];

const server = setupServer(...handlers);

export const setupServerWithErrors = () => server.use(...errorHandlers);

export default server;
