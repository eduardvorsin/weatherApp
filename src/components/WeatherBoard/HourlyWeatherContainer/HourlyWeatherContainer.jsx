import PropTypes from 'prop-types';
import { useMemo, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import fetchingStatuses from '../../../redux/fetchingStatuses';
import selectHourlyWeather from '../../../redux/selectors/selectHourlyWeather/selectHourlyWeather';
import fetchHourlyWeather from '../../../redux/thunks/fetchHourlyWeather/fetchHourlyWeather';
import { normalizeHourlyWeather, normalizeHours } from '../../../utils/utils';
import ErrorInfo from '../../UI/ErrorInfo/ErrorInfo';
import Spinner from '../../UI/Spinner/Spinner';
import Table from '../../UI/Table/Table';
import HourlyWeather from '../HourlyWeather/HourlyWeather';

export default function HourlyWeatherContainer({
  className,
  location,
  testId,
}) {
  const dispatch = useDispatch();
  const {
    data,
    loadingStatus,
  } = useSelector(selectHourlyWeather);

  useEffect(() => {
    if (location.latitude && location.longitude) {
      dispatch(fetchHourlyWeather(location));
    }
  }, [location, dispatch]);

  const content = useMemo(() => normalizeHourlyWeather(data), [data]);
  const hours = useMemo(() => normalizeHours(content), [content]);

  return (
    <section
      className={`hourly-weather__section hourly-weather-section ${className}`}
      data-testid={testId}
    >
      <h2 className="hourly-weather-section__title sr-only">
        Погода на ближайщие 24 часа
      </h2>

      {loadingStatus === fetchingStatuses.loading
        && (
          <Spinner
            size={207}
          />
        )}

      {loadingStatus === fetchingStatuses.error
        && (
          <ErrorInfo
            message="Не удалось загрузить данные о погоде на ближайщие 24 часа"
            headingLevel={3}
          />
        )}

      {loadingStatus === fetchingStatuses.resolved
        && (
          <Table
            headerCells={hours}
            dataRows={content}
            renderInDataCell={
              (_, value) => (
                <HourlyWeather
                  temperature={value.temperature}
                  weatherCode={value.weatherCode}
                />
              )
            }
          />
        )}
    </section>
  );
}

HourlyWeatherContainer.defaultProps = {
  className: '',
  location: {},
  testId: '',
};

HourlyWeatherContainer.propTypes = {
  className: PropTypes.string,
  testId: PropTypes.string,
  location: PropTypes.shape({
    latitude: PropTypes.number,
    longitude: PropTypes.number,
  }),
};
