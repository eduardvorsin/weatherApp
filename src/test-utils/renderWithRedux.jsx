import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createReduxStore } from '../redux/store';

export default function renderWithRedux(component, initialState) {
  const store = createReduxStore(initialState);

  return render(
    <Provider store={store}>
      {component}
    </Provider>,
  );
}
