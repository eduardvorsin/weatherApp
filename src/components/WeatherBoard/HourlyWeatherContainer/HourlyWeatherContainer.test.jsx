import { screen, waitForElementToBeRemoved } from '@testing-library/react';
import renderWithRedux from '../../../test-utils/renderWithRedux';
import errorState from '../../../__mocks__/errorState';
import loadingState from '../../../__mocks__/loadingState';
import mockLocation from '../../../__mocks__/location';
import resolvedState from '../../../__mocks__/resolvedState';
import server, { setupServerWithErrors } from '../../../__mocks__/server';
import HourlyWeatherContainer from './HourlyWeatherContainer';

const timeRegexp = /^\d{1,2}:\d{1,2}$/;
const temperatureRegexp = /(-)?\d+(.\d)?°/;
const weatherStatusRegexp = /ясно|пасмурно|туман|дождь|снег|гроза/;

describe('HourlyWeatherContainer component tests', () => {
  beforeAll(() => {
    jest.useFakeTimers('modern');
    jest.setSystemTime(new Date('2022-12-27T09:00:00'));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('renders correctly', () => {
    renderWithRedux(<HourlyWeatherContainer
      className="test-class"
      location={{}}
      testId="hourly-weather"
    />);

    expect(screen.getByTestId('hourly-weather')).toBeInTheDocument();
  });

  it('renders with error when loadingStatus equals "error"', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
      />,
      errorState,
    );

    expect(screen.queryAllByText(timeRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(temperatureRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(weatherStatusRegexp)).toHaveLength(0);
    expect(screen.getByText(/не удалось загрузить данные о погоде на ближайщие 24 часа/i)).toBeInTheDocument();
  });

  it('renders with loading spinner when loadingStatus equals "loading"', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
      />,
      loadingState,
    );

    expect(screen.getByTitle('loading spinner')).toBeInTheDocument();
  });

  it('renders with data when loadingStatus equals "resolved"', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
      />,
      resolvedState,
    );

    expect(screen.queryAllByText(timeRegexp)).toHaveLength(24);
    expect(screen.queryAllByText(temperatureRegexp)).toHaveLength(24);
    expect(screen.queryAllByText(weatherStatusRegexp)).toHaveLength(24);
  });

  it('the classname is correctly assigned to the element', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
        testId="hourly-weather"
      />,
    );

    expect(screen.getByTestId('hourly-weather')).toHaveClass('test-class');
  });

  it('snapshot without anything', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
        testId="hourly-weather"
      />,
    );
    expect(screen.getByTestId('hourly-weather')).toMatchSnapshot();
  });

  it('snapshot with mock error', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
        testId="hourly-weather"
      />,
      errorState,
    );

    expect(screen.getByTestId('hourly-weather')).toMatchSnapshot();
  });

  it('snapshot with mock data', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
        testId="hourly-weather"
      />,
      resolvedState,
    );

    expect(screen.getByTestId('hourly-weather')).toMatchSnapshot();
  });

  it('snapshot with a spinner', () => {
    renderWithRedux(
      <HourlyWeatherContainer
        className="test-class"
        location={{}}
        testId="hourly-weather"
      />,
      loadingState,
    );

    expect(screen.getByTestId('hourly-weather')).toMatchSnapshot();
  });
});

describe('HourlyWeatherContainer component itegration tests', () => {
  beforeAll(() => {
    jest.useFakeTimers({
      doNotFake: [
        'hrtime',
        'nextTick',
        'performance',
        'queueMicrotask',
        'requestAnimationFrame',
        'cancelAnimationFrame',
        'requestIdleCallback',
        'cancelIdleCallback',
        'setImmediate',
        'clearImmediate',
        'setInterval',
        'clearInterval',
        'setTimeout',
        'clearTimeout',
      ],
    });
    jest.setSystemTime(new Date('2022-12-27T09:00:00'));

    server.listen({ onUnhandledRequest: 'error' });
  });
  afterEach(() => server.resetHandlers());
  afterAll(() => {
    jest.useRealTimers();

    server.close();
  });

  it('the loading status changes to the error status', async () => {
    setupServerWithErrors();

    renderWithRedux(
      <HourlyWeatherContainer
        location={mockLocation}
      />,
    );

    expect(screen.getByTitle('loading spinner')).toBeInTheDocument();
    expect(screen.queryAllByText(timeRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(temperatureRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(weatherStatusRegexp)).toHaveLength(0);

    await waitForElementToBeRemoved(screen.getAllByTitle('loading spinner'));

    expect(screen.queryAllByText(timeRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(temperatureRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(weatherStatusRegexp)).toHaveLength(0);
    expect(screen.getByText(/не удалось загрузить данные о погоде на ближайщие 24 часа/i)).toBeInTheDocument();
  });

  it('the loading status changes to the resolved status', async () => {
    renderWithRedux(
      <HourlyWeatherContainer
        location={mockLocation}
      />,
    );

    expect(screen.getByTitle('loading spinner')).toBeInTheDocument();
    expect(screen.queryAllByText(timeRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(temperatureRegexp)).toHaveLength(0);
    expect(screen.queryAllByText(weatherStatusRegexp)).toHaveLength(0);

    await waitForElementToBeRemoved(screen.getAllByTitle('loading spinner'));

    expect(screen.queryAllByText(timeRegexp)).toHaveLength(24);
    expect(screen.queryAllByText(temperatureRegexp)).toHaveLength(24);
    expect(screen.queryAllByText(weatherStatusRegexp)).toHaveLength(24);
    expect(screen.queryByText(/не удалось загрузить данные о погоде на ближайщие 24 часа/i)).not.toBeInTheDocument();
  });
});
