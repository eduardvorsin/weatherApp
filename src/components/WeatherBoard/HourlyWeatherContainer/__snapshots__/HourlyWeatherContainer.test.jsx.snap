// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlyWeatherContainer component tests snapshot with a spinner 1`] = `
<section
  class="hourly-weather__section hourly-weather-section test-class"
  data-testid="hourly-weather"
>
  <h2
    class="hourly-weather-section__title sr-only"
  >
    Погода на ближайщие 24 часа
  </h2>
  <div
    aria-label="loading spinner"
    class="spinner "
    style="--size: 207px;"
    title="loading spinner"
  >
    <div
      class="spinner__ring"
      style="--ring-size: 165.60000000000002px;"
    />
    <div
      class="spinner__ring"
      style="--ring-size: 165.60000000000002px;"
    />
    <div
      class="spinner__ring"
      style="--ring-size: 165.60000000000002px;"
    />
    <div
      class="spinner__ring"
      style="--ring-size: 165.60000000000002px;"
    />
  </div>
</section>
`;

exports[`HourlyWeatherContainer component tests snapshot with mock data 1`] = `
<section
  class="hourly-weather__section hourly-weather-section test-class"
  data-testid="hourly-weather"
>
  <h2
    class="hourly-weather-section__title sr-only"
  >
    Погода на ближайщие 24 часа
  </h2>
  <div
    class="table-overlay "
    data-testid=""
  >
    <div
      class="table-container"
    >
      <table
        class="table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="table__row"
          >
            <th
              class="table__head-cell"
            >
              9:00
            </th>
            <th
              class="table__head-cell"
            >
              10:00
            </th>
            <th
              class="table__head-cell"
            >
              11:00
            </th>
            <th
              class="table__head-cell"
            >
              12:00
            </th>
            <th
              class="table__head-cell"
            >
              13:00
            </th>
            <th
              class="table__head-cell"
            >
              14:00
            </th>
            <th
              class="table__head-cell"
            >
              15:00
            </th>
            <th
              class="table__head-cell"
            >
              16:00
            </th>
            <th
              class="table__head-cell"
            >
              17:00
            </th>
            <th
              class="table__head-cell"
            >
              18:00
            </th>
            <th
              class="table__head-cell"
            >
              19:00
            </th>
            <th
              class="table__head-cell"
            >
              20:00
            </th>
            <th
              class="table__head-cell"
            >
              21:00
            </th>
            <th
              class="table__head-cell"
            >
              22:00
            </th>
            <th
              class="table__head-cell"
            >
              23:00
            </th>
            <th
              class="table__head-cell"
            >
              0:00
            </th>
            <th
              class="table__head-cell"
            >
              1:00
            </th>
            <th
              class="table__head-cell"
            >
              2:00
            </th>
            <th
              class="table__head-cell"
            >
              3:00
            </th>
            <th
              class="table__head-cell"
            >
              4:00
            </th>
            <th
              class="table__head-cell"
            >
              5:00
            </th>
            <th
              class="table__head-cell"
            >
              6:00
            </th>
            <th
              class="table__head-cell"
            >
              7:00
            </th>
            <th
              class="table__head-cell"
            >
              8:00
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        >
          <tr
            class="table__row"
          >
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="ясно"
                  class="hourly-weather__icon"
                  src="/assets/icons/sunny.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -4.5
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  ясно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -4.4
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -4.3
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="снег"
                  class="hourly-weather__icon"
                  src="/assets/icons/snow.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -4.5
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  снег
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="снег"
                  class="hourly-weather__icon"
                  src="/assets/icons/snow.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.1
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  снег
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.6
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.8
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -6
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -6.5
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -6.8
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="ясно"
                  class="hourly-weather__icon"
                  src="/assets/icons/sunny.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -7
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  ясно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -6.4
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="снег"
                  class="hourly-weather__icon"
                  src="/assets/icons/snow.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.3
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  снег
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="дождь"
                  class="hourly-weather__icon"
                  src="/assets/icons/rain.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -4.3
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  дождь
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="ясно"
                  class="hourly-weather__icon"
                  src="/assets/icons/sunny.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -3.6
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  ясно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="ясно"
                  class="hourly-weather__icon"
                  src="/assets/icons/sunny.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -3.4
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  ясно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -3.7
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -4.4
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -4.9
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="снег"
                  class="hourly-weather__icon"
                  src="/assets/icons/snow.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.3
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  снег
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="снег"
                  class="hourly-weather__icon"
                  src="/assets/icons/snow.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.6
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  снег
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.8
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="пасмурно"
                  class="hourly-weather__icon"
                  src="/assets/icons/cloudy.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.7
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  пасмурно
                </p>
              </div>
            </td>
            <td
              class="table__data-cell"
            >
              <div
                class="hourly-weather "
                data-testid=""
              >
                <img
                  alt="ясно"
                  class="hourly-weather__icon"
                  src="/assets/icons/sunny.svg"
                />
                <p
                  class="hourly-weather__temperature"
                >
                  -5.5
                  °
                </p>
                <p
                  class="hourly-weather__status"
                >
                  ясно
                </p>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</section>
`;

exports[`HourlyWeatherContainer component tests snapshot with mock error 1`] = `
<section
  class="hourly-weather__section hourly-weather-section test-class"
  data-testid="hourly-weather"
>
  <h2
    class="hourly-weather-section__title sr-only"
  >
    Погода на ближайщие 24 часа
  </h2>
  <div
    class="error-info  "
    data-testid=""
    role="alert"
  >
    <h3
      class="error-info__description"
    >
      Не удалось загрузить данные о погоде на ближайщие 24 часа
    </h3>
  </div>
</section>
`;

exports[`HourlyWeatherContainer component tests snapshot without anything 1`] = `
<section
  class="hourly-weather__section hourly-weather-section test-class"
  data-testid="hourly-weather"
>
  <h2
    class="hourly-weather-section__title sr-only"
  >
    Погода на ближайщие 24 часа
  </h2>
</section>
`;
